### ISSUE TITLE
[PrimeFaces showcase] chat example is not working

### ISSUE THREAD

**Issue by hitochan777** [primefaces/primefaces#2503]
## 1) Environment
- PrimeFaces version: 6.1
- Atmosphere version: 2.4.12
- Does it work on the newest released PrimeFaces version? Version?
No
- Application server + version: GlassFish 4.1.2
- Affected browsers:
Tested with Chrome only 

## 2) Expected behavior
I am running [Chat example in showcase](https://www.primefaces.org/showcase/push/chat.xhtml).
1. Users can login
2. Users are notified when new users log in

## 3) Actual behavior
1. Nothing happens when I click "Login" button. This is because `init` annotated with `@PostConstruct` is not called and `ChatView.users` is `null`. ( I fixed the above problem by assigning the instance as the instance variable is declared. I am not sure if this is the right way to fix it. )
2. Users are not notified when new users log in. 
In ChatResource.java

```
    @OnClose
    public void onClose(RemoteEndpoint r, EventBus eventBus) {
        ChatUsers users= (ChatUsers) ctx.getAttribute("chatUsers");
        users.remove(username);
         
        eventBus.publish(room + "/*", new Message(String.format("%s has left the room", username), true));
    }
``` 
This is because `eventBus` is `null`.

## 4) Steps to reproduce
I used exactly the same code as [the chat example in PrimeFaces showcase](https://www.primefaces.org/showcase/push/chat.xhtml), except that I enabled the input textbox and button in the `chat.xhtml`.

## 5) Sample XHTML
See [chat.xhtml](https://www.primefaces.org/showcase/push/chat.xhtml), and enable the textbox and button.

## 6) Sample bean
See [Chat example](https://www.primefaces.org/showcase/push/chat.xhtml)

## 7) Question
1. What are the right way to fix the above problems?
2. Is it possible to inject CDI bean (For example, a class annotated with `@javax.enterprise.context.ApplicationScoped`) to the class annotated with `@PushEndPoint`? It seems that we cannot inject either JSF beans and CDI beans. And `ServletContext.getAttribute` can only access JSF beans. An example is as follows.

ChatResource.java
```
@PushEndpoint("/{room}/{user}")
@Singleton
public class ChatResource {
 
    @PathParam("room")
    private String room;
 
    @PathParam("user")
    private String username;
 
    @Inject
    private ChatUsers users;
 
   ...
}
```

ChatUsers.java
```
...
import javax.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class ChatUsers implements Serializable {
     
    private List<String> users;
     
    @PostConstruct
    public void init() {
        users = new ArrayList<String>();
    }
    ...
}
```

**Comment by tandraschko** [comment]
If you could provide a PR to fix it, it would be great. 
I don't use PrimePush and don't have to to look at it.

**Comment by hitochan777** [comment]
@tandraschko OK, I will try.

**Comment by mismanc** [comment]
I have the same issue:

    @OnOpen
    public void onOpen(RemoteEndpoint r, EventBus eventBus) {
        //logger.info("OnOpen {}", r);
        String user = r.pathSegments(2);
        String ru = r.pathSegments(1);
    
        if (eventBus != null) {
            eventBus.publish("chat/*", new ChatMessage(user,"Entered Room " + user,true));
        } else {
            System.out.println("EVENT BUS IS NULL");
            eventBus = EventBusFactory.getDefault().eventBus();
            eventBus.publish("chat/*", new ChatMessage(String.format("%s has entered the room '%s'", user, ru), true));
        }
    }

    @OnClose
    public void onClose(RemoteEndpoint r, EventBus eventBus) {
        String u = r.pathSegments(2);
        String ru = r.pathSegments(1);
        if (eventBus != null) {
            eventBus.publish("chat/*", new ChatMessage(String.format("%s get out the room '%s'", u, ru), true));
        } else {
            System.out.println("EVENT BUS IS NULL");
            eventBus = EventBusFactory.getDefault().eventBus();
            eventBus.publish("chat/*", new ChatMessage(String.format("%s get out the room '%s'", u, ru), true));
        }
    }


I have tried different versions of `atmosphere-runtime`(2.4.6, 2.4.9, 2.4.14). it enters always `else` block. So the `eventBus` is coming null. And I am not sure, it is the best way to handle the issue.

**Comment by tandraschko** [comment]
PrimeFaces Push will be deprecated in 6.2 and removed in 6.3 ( see https://www.primefaces.org/primefaces-6-2-roadmap/)
Therefor i will close this issue as won't fix. If you will provide a PR to fix this issue until 6.2 RC1, i can merge it.
